packages/framework/src/Command/FixExtendedClassesAnnotationsCommand.php

```diff
@@ -0,0 +1,209 @@
+            ->setHelp('What does the command do exactly? (When dry-run is set to false)
+- Replaces the framework with the project annotations in all project files when there exists a project extension of a given framework class.
+- Adds @property annotations to project classes when there exists a property in parent class that is extended in the project.
+- Adds @method annotations to project classes when there exists a method in parent class that returns an instance of a class that is extended in the project.');
+    }
+
+    /**
+     * @param string $projectRootDirectory
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry $classExtensionRegistry
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\PropertyAnnotationsFactory $propertyAnnotationsFactory
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\MethodAnnotationsFactory $methodAnnotationsAdder
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\AnnotationsReplacer $annotationsReplacer
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\AnnotationsReplacementsMap $annotationsReplacementsMap
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\AnnotationsAdder $annotationsAdder
+     */
+    public function __construct(
+        string $projectRootDirectory,
+        ClassExtensionRegistry $classExtensionRegistry,
+        PropertyAnnotationsFactory $propertyAnnotationsFactory,
+        MethodAnnotationsFactory $methodAnnotationsAdder,
+        AnnotationsReplacer $annotationsReplacer,
+        AnnotationsReplacementsMap $annotationsReplacementsMap,
+        AnnotationsAdder $annotationsAdder
+    ) {
+        parent::__construct();
+        $this->projectRootDirectory = $projectRootDirectory;
+        $this->classExtensionRegistry = $classExtensionRegistry;
+        $this->annotationsReplacer = $annotationsReplacer;
+        $this->annotationsReplacementsMap = $annotationsReplacementsMap;
+        $this->propertyAnnotationsFactory = $propertyAnnotationsFactory;
+        $this->methodAnnotationsAdder = $methodAnnotationsAdder;
+        $this->annotationsAdder = $annotationsAdder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $symfonyStyle = new SymfonyStyle($input, $output);
+        $isDryRun = (bool)$input->getOption(static::DRY_RUN);
+        $filesForReplacingAnnotations = $this->replaceFrameworkWithProjectAnnotations($isDryRun);
+        if (!empty($filesForReplacingAnnotations)) {
+            if ($isDryRun) {
+                $symfonyStyle->error(sprintf("Following files need fixing annotations:\n%s", implode($filesForReplacingAnnotations, "\n")));
+            } else {
+                $symfonyStyle->note(sprintf("Annotations were fixed in the following files:\n%s", implode($filesForReplacingAnnotations, "\n")));
```

Comment:
list in note seems ok to me, but please use arrays to improve readability (comment above about dark magic)

Reply:
02408bf28